src/Analysis/Ast/Impl/Types/Definitions/IPythonClassType.cs

```diff
@@ -36,6 +36,11 @@ public interface IPythonClassType : IPythonType, ILocatedMember {
         /// </summary>
         IReadOnlyList<IPythonType> Bases { get; }
 
+        /// <summary>
+        /// If the class contains an abstract class on its inheritance chain
+        /// </summary>
+        bool IsDerivedFromAbstractClass { get; set;  }
```

Comment:
Probably should be an extension as mentioned above.

Reply:
so take for example:

```
class A:
  @abstractmethod
  def tmp(self): 
     pass

class B(A):
  def tmp(self):
     pass

class C(B):
  @abstractmethod 
  def tmp2(self):
    pass
```

Class C is still considered abstract even though it inherits a class that is not abstract. Therefore I say that C is derived from an abstract class, which basically means C (and any descendants) can become abstract, even if they do not inherit directly from abstract classes.